### PR TITLE
Remove redundant slashes in URI-string before it's URI will be resolve.

### PR DESCRIPTION
**Pull request by daggerok**
See more about issue here: https://github.com/spring-projects/spring-boot/issues/9133

### CODE REVIEW

**Comment by snicoll** [comment]
@daggerok as I've asked you on the issue you've opened in Spring Boot, we need a Jira issue in our tracker for this change. Thanks!

**Comment by daggerok** [comment]
this one is ok? 
https://jira.spring.io/browse/SPR-15529

**Comment by daggerok** [comment]
I reviewed little bit and understand, that this PR wont fix an issue

File: spring-web/src/main/java/org/springframework/http/server/reactive/ReactorServerHttpRequest.java [hunk 1]
```diff
@@ -58,10 +60,21 @@ public ReactorServerHttpRequest(HttpServerRequest request, NettyDataBufferFactor
...
 
+	/**
+	 * Remove redundant slashes from request URI-string
+	 *
+	 * <p>
+	 *   http://example.com/// -> http://example.com/
+	 * </p>
+	 */
+	private static String squashSlashes(String uri) {
+		return isNull(uri) ? uri : uri.replaceAll("/{2,}", "/");
```
Comment (Shredder121) [inline spring-web/src/main/java/org/springframework/http/server/reactive/ReactorServerHttpRequest.java]: Wouldn't this collapse `http://example.com///` to `http:/example.com/`?

Comment (daggerok) [inline spring-web/src/main/java/org/springframework/http/server/reactive/ReactorServerHttpRequest.java]: you are talking about URL, but npt URI, with actually i'm replacing, so answering on your question: no, it wouldn't collapse `http://x.y.z///` to `http:/x.y.z/`, instead, it should collapse into `http://x.y.z/`

**Comment by snicoll** [comment]
@daggerok is that PR still relevant?

**Comment by rstoyanchev** [comment]
This evolved from https://jira.spring.io/browse/SPR-15529 to https://jira.spring.io/browse/SPR-15560 and where the actual issue was not the duplicate slash but a bug in how `ReactorServerHttpRequest` created the URI. So this can be closed.

File: spring-web/src/main/java/org/springframework/http/server/reactive/ReactorServerHttpRequest.java (hunk 1 above)
Comment (luyuanwan) [inline spring-web/src/main/java/org/springframework/http/server/reactive/ReactorServerHttpRequest.java]: I think function squashSlashes should in some tool utility class. :)

**Pull request closed without merging**